ocaml/toplevel/genprintval.ml

```diff
@@ -261,6 +261,7 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
       | Void -> Print_as "<void>"
       | Any | Any_non_null -> Print_as "<any>"
       | Float64 | Float32 | Bits32 | Bits64 | Word -> Print_as "<abstr>"
+      | Product _ -> Print_as "<unboxed product>"
```

Comment:
I agree, but I think I'll just make a ticket for now.  Note that all other unboxed types are also not printed well here, so this isn't new.  And I think it's hard to actually reach this case, since you can't make a top-level binding to an unboxed product or put one in a structure.